src/cargo/util/rustc.rs

```diff
@@ -42,11 +44,28 @@ impl Rustc {
             triple.to_string()
         };
 
+        let mut cmd = util::process(&path);
```

Comment:
I think we can do both feature detects at once if we grep `rustc --help` for `cap-lints` and `metadata`.